Mathlib/Topology/Algebra/Order/IntermediateValue.lean

```diff
@@ -644,3 +648,122 @@ theorem ContinuousOn.surjOn_of_tendsto' {f : α → δ} {s : Set α} [OrdConnect
     (htop : Tendsto (fun x : s => f x) atTop atBot) : SurjOn f s univ :=
   @ContinuousOn.surjOn_of_tendsto α _ _ _ _ δᵒᵈ _ _ _ _ _ _ hs hf hbot htop
 #align continuous_on.surj_on_of_tendsto' ContinuousOn.surjOn_of_tendsto'
+
+/-- Suppose `f : [a, b] → δ` is
+continuous and injective. Then `f` is strictly monotone (increasing)
+if `f(a) ≤ f(b)`.-/
+theorem ContinuousOn.StrictMonoOn_of_InjOn_Icc {a b : α} {f : α → δ}
+    (hab : a ≤ b) (hfab : f a ≤ f b)
+    (hf_c : ContinuousOn f (Icc a b)) (hf_i : InjOn f (Icc a b)) :
+    StrictMonoOn f (Icc a b) := by
```

Comment:
I think this theorem would be a lot less headache to prove if you did it for types first. Then you can prove the version for sets using the one for types. Note: this is how the intermediate value theorem is proved as well.

The right generality might be something like this.
```
theorem Continuous.strictMono_of_injective {α : Type*} [ConditionallyCompleteLinearOrder α] [TopologicalSpace α] [OrderTopology α] [DenselyOrdered α] {δ : Type*} [LinearOrder δ] [TopologicalSpace δ] [OrderClosedTopology δ] {f : C(α, δ)} (hf₁ : f ⊥ ≤ f ⊤) (hf₂ : Function.Injective f) : StrictMono f := sorry
```
If you do it this way, then you don't have to continue worry about whether or not your elements are in the interval or not. Then to prove the version for intervals, you just use `ContinuousOn.restrict`.

Reply:
Thank you very much! I add `Continuous.strictMono_of_injective` now. However, I find that the order on a closed interval is not necessarily a conditionally complete linear order. So I cannot deduce `ContinuousOn.StrictMonoOn_of_InjOn_Icc` from `Continuous.strictMono_of_injective` yet.

Reply:
I cannot deduce `ContinuousOn.StrictMonoOn_of_InjOn_Icc` from `Continuous.strictMono_of_injective` in Lean. It is very strange, and I need help: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/How.20to.20tell.20Lean.20preorders.20from.20different.20sources.20are.20same.3F

Reply:
Many thanks to @alreadydone ! I can deduce `ContinuousOn.StrictMonoOn_of_InjOn_Icc` from `Continuous.strictMono_of_injective` now.